Assets/Scripts/DeltaDNAMediationListener.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Mediation;
+using DeltaDNA;
+
+// var handler = MediationListener();
+// adapter.LoadInterstitial(handler, null);
+
+// TODO: consider alternate names.
+public class DeltaDNAMediationListener: IUnityMediationAdUnitListener
+{
+    private IUnityAdUnit _adUnit;
+    private ImageMessage _imageMessage;
+
+    private bool shouldShowPromo = false;
+
+    private bool _ready = false;
+
+    public void Show() {
+        if (!_ready) {
+            return;
+        }
+
+        if (_adUnit == null && _imageMessage == null) {
+            return;
+        }
+
+        if (_imageMessage != null) {
```

Comment:
we may wants to call deltaDNA on deltaDNA promo shown callback too. just like onShown for mediation.